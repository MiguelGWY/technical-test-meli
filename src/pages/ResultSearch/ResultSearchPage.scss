.container-products {
  display: flex;
  flex-direction: column;
  align-items: center;

  &__container-breadCrumbs {
    width: 780px;
    height: 50px;
  }

  &__container-list {
    width: 750px;
    padding: 15px;
    background-color: white;
    border-radius: 5px;
  }

  &__ol {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__link {
    text-decoration: none;
  }

  &__li {
    background-color: white;
  }

}
