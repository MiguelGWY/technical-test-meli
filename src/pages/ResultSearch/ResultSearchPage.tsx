import { useEffect, useState } from "react";
import { getProducts } from "../../api/services/products.service";
import { HeaderComponent } from "../../components/Header/HeaderComponent";
import { BreadCrumbs } from "../../components/BreadCrumbs/BreadCrumbsComponent";
import { ProductCardComponent } from "../../components/ProductCard/ProductCardComponent";
import { Link, useLocation } from "react-router-dom";
import "./ResultSearchPage.scss";

export const ResultSearchPage = () => {
  const nameSpace = "container-products";
  const location = useLocation();
  const searchParams = new URLSearchParams(location.search).get("search");
  const [products, setProducts] = useState<ItemSearchResponse | undefined>();

  useEffect(() => {
    const fetchProductsData = async () => {
      try {
        const productSearch = await getProducts(searchParams || "");
        setProducts(productSearch);
      } catch (error) {
        console.error("Se presentó un error en la petición", error);
      }
    };

    fetchProductsData();
  }, [searchParams]);

  const renderProductList = () => {
    return (
      <ol className={`${nameSpace}__ol`}>
        {products?.items?.map((product) => (
          <li key={product.id} className={`${nameSpace}__li`}>
            <Link to={`/items/${product.id}`} className={`${nameSpace}__link`}>
              <ProductCardComponent {...product} />
            </Link>
          </li>
        ))}
      </ol>
    );
  };

  return (
    <>
      <HeaderComponent />
      <section className={nameSpace}>
        <div className={`${nameSpace}__container-breadCrumbs`}>
          <BreadCrumbs categories={products?.categories || []} />
        </div>
        <div className={`${nameSpace}__container-list`}>
          {products && renderProductList()}
        </div>
      </section>
    </>
  );
};
