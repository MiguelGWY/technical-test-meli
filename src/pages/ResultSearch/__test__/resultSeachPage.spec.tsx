import "@testing-library/jest-dom";
import { render, screen, waitFor } from "@testing-library/react";
import { ResultSearchPage } from "../ResultSearchPage";
import { MemoryRouter } from "react-router-dom";
import { getProducts } from "../../../api/services/products.service";

// Mock de productos
jest.mock("../../../api/services/products.service");

describe("ResultSearchPage", () => {
  it("should render ResultSearchPage", () => {
    const { container } = render(
      <MemoryRouter>
        <ResultSearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
  
  it("should render HeaderComponent and BreadCrumbsComponent", async () => {
    const mockProducts = {
      items: [
        { id: 1, title: "Product 1", price: 10 },
        { id: 2, title: "Product 2", price: 20 },
      ],
      categories: ["Category 1", "Category 2"],
    };

    (getProducts as jest.Mock).mockResolvedValueOnce(mockProducts);

    render(
      <MemoryRouter>
        <ResultSearchPage />
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByTestId("header-component")).toBeInTheDocument();
      expect(screen.getByTestId("container-breadcrumbs")).toBeInTheDocument();
    });
  });

  it("should render product list", async () => {
    // Mock de datos de productos
    const mockProducts = {
      items: [
        { id: 1, title: "Product 1", price: 10 },
        { id: 2, title: "Product 2", price: 20 },
      ],
      categories: ["Category 1", "Category 2"],
    };

    (getProducts as jest.Mock).mockResolvedValueOnce(mockProducts);

    render(
      <MemoryRouter>
        <ResultSearchPage />
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText("Product 1")).toBeInTheDocument();
      expect(screen.getByText("Product 2")).toBeInTheDocument();
    });
  });

});
