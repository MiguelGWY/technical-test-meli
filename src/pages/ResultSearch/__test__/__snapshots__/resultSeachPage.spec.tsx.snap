// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultSearchPage should render ResultSearchPage 1`] = `
<div>
  <header
    class="nav-header"
    data-testid="header-component"
  >
    <div
      class="nav-header__container"
    >
      <div
        class="nav-header__logo"
      >
        <button
          aria-label="Regresar a la home"
          class="nav-header__redirect-button"
        >
          <img
            alt="logo mercadolibre"
            src="https://http2.mlstatic.com/frontend-assets/ml-web-navigation/ui-navigation/6.6.43/mercadolibre/logo_large_25years_v2.png"
          />
        </button>
      </div>
      <div
        class="nav-header__search"
      >
        <form
          class="nav-header__form"
          data-testid="form-input"
        >
          <label
            class="nav-header__label"
          >
            Ingresa lo que quieras encontrar
          </label>
          <input
            class="nav-header__search-input"
            placeholder="Nunca dejes de buscar"
            type="text"
            value=""
          />
        </form>
      </div>
    </div>
  </header>
  <section
    class="container-products"
  >
    <div
      class="container-products__container-breadCrumbs"
    />
    <div
      class="container-products__container-list"
    />
  </section>
</div>
`;
